### PR TITLE
Watchdog: Use 'reset' as default when no device is attached

### PR DESCRIPTION
**Pull request by skobyda**
Fixes https://github.com/cockpit-project/cockpit-machines/issues/1249

### Before

[Screencast from 2023-10-12 12-25-24.webm](https://github.com/cockpit-project/cockpit-machines/assets/42733240/7d0aabc2-90aa-4d2b-b365-6783b6834264)

### After

[Screencast from 2023-10-12 12-26-09.webm](https://github.com/cockpit-project/cockpit-machines/assets/42733240/65a8d668-9e87-4db8-af6a-2817c861599f)

### CODE REVIEW

File: src/components/vm/overview/watchdog.jsx [hunk 1]
```diff
@@ -110,6 +112,11 @@ export const WatchdogModal = ({ vm, isWatchdogAttached, idPrefix }) => {
     };
 
     const save = (coldplug) => {
+        if (!watchdogAction) {
+            setValidationMessage(_("You need to select the action"));
+            return;
```
Comment (cockpituous) [inline src/components/vm/overview/watchdog.jsx:117]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1256-20231012-104029-22248714-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by martinpitt** [comment]
The "you need to select the action" string is some bikeshed territory, but my gut feeling is that the "Add" button could simply be disabled while there is nothing selected?

**Comment by skobyda** [comment]
> The "you need to select the action" string is some bikeshed territory, but my gut feeling is that the "Add" button could simply be disabled while there is nothing selected?

Do you mean that Add button would be disabled immediately when dialog is opened? That this is not good user experience. I.e. user only opens dialog, they didn't do anything yet, and we are basically telling them that something is already wrong by having a disabled button. That's confusing. I remember we had discussion about this a year ago with @garrett  when implementing watchdog.

Or do you mean that user would click on "Add" and when nothing is selected, it would get disabled, but "you need to select the action" message is not shown? I also find that a bit confusing, since we are disabling the button as a result of user action, but not providing any feedback.

**Comment by jelly** [comment]
> > The "you need to select the action" string is some bikeshed territory, but my gut feeling is that the "Add" button could simply be disabled while there is nothing selected?
> 
> Do you mean that Add button would be disabled immediately when dialog is opened? That this is not good user experience. I.e. user only opens dialog, they didn't do anything yet, and we are basically telling them that something is already wrong by having a disabled button. That's confusing. I remember we had discussion about this a year ago with @garrett when implementing watchdog.
> 
> Or do you mean that user would click on "Add" and when nothing is selected, it would get disabled, but "you need to select the action" message is not shown? I also find that a bit confusing, since we are disabling the button as a result of user action, but not providing any feedback.

I think that is the correct approach yes, disabling until a valid choice has been made. And not showing the warning that you have to make a selection.

**Comment by martinpitt** [comment]
> Do you mean that Add button would be disabled immediately when dialog is opened?

Yes.

> That this is not good user experience. I.e. user only opens dialog, they didn't do anything yet, and we are basically telling them that something is already wrong by having a disabled button

But with an enabled button we'd then show the error message, so it's the same result -- the only valid action without picking a value is "Cancel".

**Comment by skobyda** [comment]
> But with an enabled button we'd then show the error message, so it's the same result -- the only valid action without picking a value is "Cancel".

Maybe we could at least have a Tooltip when "Add" button is hovered over? @garrett WDYT?

**Comment by garrett** [comment]
Whooooooah... I'd suggest we step back a little. :grin: 

This is a list of radios. Radios, by default, should have one preselected — and that's almost always the first item. The fact that there isn't one preselected here is the bug. If the person using Cockpit Machines encounters this modal and doesn't want the first item — or any other — then they can always click cancel.

So: Please make the first default when adding a watchdog. If first isn't a good default, then we should reorder the list so it is.

(As it's a list of radios, as long as one is selected, then you cannot unselect, so someone would then never encounter this issue, and we'll never have to explain it.)

I think that'll solve all these issues. Right?

---

TL;DR: Preselect the first radio item, and that should solve this.

**Comment by skobyda** [comment]
Updated the PR, so the "Reset" option is pre-selected as default when no watchdog device is attached yet:

[Screencast from 2023-10-13 12-23-18.webm](https://github.com/cockpit-project/cockpit-machines/assets/42733240/48f27091-7723-4658-a745-9ab7efda8523)

**Review by martinpitt — APPROVED** [review]
Works for me, thanks!
